### ISSUE TITLE
Reimplement codeformatter "rules" as Roslyn analyzers/fixers

### ISSUE THREAD

**Issue by lgolding** [dotnet/codeformatter#137]
This will allow the rules to be used in VS as well.

**Comment by Priya91** [comment]
Sounds good.

**Comment by twsouthwick** [comment]
@Igolding Are you still working on this?

**Comment by lgolding** [comment]
Yes, kind of a long term effort :) + @michaelcfanning.

**Comment by michaelcfanning** [comment]
Yes, in flight. :) We need to reauthor the range of existing checks as analyzers. We should enable codeformatter to drive arbitrary analyzers. We need to good configuration story for the analyzers (as these checks are mostly highly configurable).

**Comment by stebet** [comment]
I'm very much looking forward to this. Is there a repo we can contribute into to help with this task?

**Comment by jaredpar** [comment]
@stebet this work is going on in the anfix branch

https://github.com/dotnet/codeformatter/tree/anfix

**Comment by dnperfors** [comment]
I didn't know there was already work like this done, so I started my own project for analyzers: https://github.com/dnperfors/CodeStyleAnalyzer There is already a version available on NuGet
